btp/provider/resource_subaccount_destination_fragment.go

```diff
@@ -0,0 +1,363 @@
+		Name:              plan.Name,
+		ServiceInstanceID: plan.ServiceInstanceID,
+	}
+
+	diags = resp.Identity.Set(ctx, identity)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (rs *subaccountDestinationFragmentResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var plan subaccountDestinationFragmentResourceConfig
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	rawContent, diags := extractFragmentContent(plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	payload := buildPayload(plan.Name.ValueString(), rawContent)
+
+	hasServiceInstance := !plan.ServiceInstanceID.IsNull() && !plan.ServiceInstanceID.IsUnknown() && plan.ServiceInstanceID.ValueString() != ""
+
+	err := rs.updateFragment(ctx, plan, payload, hasServiceInstance, resp.Diagnostics)
+	if err != nil {
+		return
+	}
+
+	destinationFragmentDetails, err := rs.readFragment(ctx, plan, hasServiceInstance, resp.Diagnostics)
+	if err != nil {
+		return
+	}
+
+	delete(destinationFragmentDetails.Content, "FragmentName")
+
+	if !plan.DestinationFragment.IsNull() {
+		plan.DestinationFragment, diags = types.MapValueFrom(ctx, types.StringType, destinationFragmentDetails.Content)
+		resp.Diagnostics.Append(diags...)
+	}
+
+	diags = resp.State.Set(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+
+	identity := subaccountDestinationFragmentIdentityModel{
+		SubaccountID:      plan.SubaccountID,
+		Name:              plan.Name,
+		ServiceInstanceID: plan.ServiceInstanceID,
+	}
+
+	diags = resp.Identity.Set(ctx, identity)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (rs *subaccountDestinationFragmentResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state subaccountDestinationFragmentResourceConfig
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	hasServiceInstance := !state.ServiceInstanceID.IsNull() && !state.ServiceInstanceID.IsUnknown() && state.ServiceInstanceID.ValueString() != ""
+
+	if hasServiceInstance {
+		_, _, err := rs.cli.Connectivity.DestinationFragment.DeleteByServiceInstance(ctx, state.SubaccountID.ValueString(), state.Name.ValueString(), state.ServiceInstanceID.ValueString())
+		if err != nil {
+			resp.Diagnostics.AddError("API Error Deleting Destination Fragment at Service Instance Level", fmt.Sprintf("%s", err))
+			return
+		}
+	} else {
+		_, _, err := rs.cli.Connectivity.DestinationFragment.DeleteBySubaccount(ctx, state.SubaccountID.ValueString(), state.Name.ValueString())
+		if err != nil {
+			resp.Diagnostics.AddError("API Error Deleting Destination Fragment at Subaccount Level", fmt.Sprintf("%s", err))
+			return
+		}
+	}
+}
+
+func (rs *subaccountDestinationFragmentResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
```

Comment:
Missed that: Import must support both variants: classical import plus import via resource identity. Reason for that is that OpenTofu does not support imports via resource identity yet